zerver/lib/actions.py

```diff
@@ -3621,7 +3623,12 @@ def do_update_message_flags(user_profile: UserProfile,
                             operation: str,
                             flag: str,
                             messages: List[int]) -> int:
-    flagattr = getattr(UserMessage.flags, flag)
+    flags = [item for item in UserMessage.flags if item not in UserMessage.NON_API_FLAGS]
+    if flag in flags:
+        # `flags` is a list. Do not use `flags` here by mistake.
+        flagattr = getattr(UserMessage.flags, flag)
+    else:
+        raise JsonableError(_("Invalid flag: %s" % (flag,)))
```

Comment:
We should do an early-exit with `JsonableError`.  Also, I renamed `flags` to `valid_flags` so we don't need the comment.